plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/editor/OpenDeclarationFinder.java

```diff
@@ -162,48 +170,52 @@ public void onEditorActivated(EditorPartPresenter editor) {
-                            @Override
-                            public void apply(ClassContent content) throws OperationException {
-                              final VirtualFile file =
-                                  javaNodeFactory.newJarFileNode(
-                                      entry, descriptor.getLibId(), projectPath, null);
-                              editorAgent.openEditor(
-                                  file,
-                                  new OpenEditorCallbackImpl() {
-                                    @Override
-                                    public void onEditorOpened(final EditorPartPresenter editor) {
-                                      Scheduler.get()
-                                          .scheduleDeferred(
-                                              new Scheduler.ScheduledCommand() {
-                                                @Override
-                                                public void execute() {
-                                                  if (editor instanceof TextEditor) {
-                                                    ((TextEditor) editor)
-                                                        .getDocument()
-                                                        .setSelectedRange(
-                                                            LinearRange.createWithStart(
-                                                                    descriptor.getOffset())
-                                                                .andLength(0),
-                                                            true);
-                                                    editor.activate();
-                                                  }
-                                                }
-                                              });
-                                    }
-                                  });
-                            }
-                          });
-                }
+              entry -> {
+                ExternalLibrariesParameters params =
+                    dtoFactory.createDto(ExternalLibrariesParameters.class);
+                params.setNodeId(descriptor.getLibId());
+                params.setNodePath(entry.getPath());
+                params.setProjectUri(projectPath.toString());
+                extensionService
+                    .libraryNodeContentByPath(params)
+                    .then(
+                        content -> {
+                          final VirtualFile file =
+                              javaNodeFactory.newJarFileNode(
+                                  entry, descriptor.getLibId(), projectPath, null);
+                          editorAgent.openEditor(
+                              file,
+                              new OpenEditorCallbackImpl() {
+                                @Override
+                                public void onEditorOpened(final EditorPartPresenter editor) {
```

Comment:
You could make reviewer's work easier if you extracted some excerpt of code into separate method.

Reply:
Basically it is formatter changes. Also this code will remove because `Find Definition`  feature from jdt.ls will replace this one. So I don't care to do code cleanup here.

Reply:
And use lambda notation where possible.